main.tf

```diff
@@ -21,7 +21,7 @@ resource "aws_vpn_gateway" "default" {
 # https://www.terraform.io/docs/providers/aws/r/customer_gateway.html
 resource "aws_customer_gateway" "default" {
   count            = local.enabled && var.customer_gateway_ip_address != null ? 1 : 0
-  device_name      = module.this.id
+  device_name      = var.customer_gateway_device_name == "" ? module.this.id : var.customer_gateway_device_name
```

Comment:
Any reason to not use `null` for the var?

Reply:
mmm true, it could have been nulll.